src/intops/impl/inlineasm/x86.nim

```diff
@@ -0,0 +1,161 @@
+## Arithmetic operations for integers implemented in x86 GCC/Clang inline Assembly.
+
+import ../../consts
+
+when cpu64Bit and cpuX86 and compilerGccCompatible and canUseInlineAsm:
+  func carryingAdd*(a, b: uint64, carryIn: bool): (uint64, bool) {.inline.} =
+    var
+      sum = a
+      cOut: uint8
+      cInVal = if carryIn: 1'u64 else: 0'u64
+
+    asm """
+      negq %2      /* Sets CF=1 if cInVal==1, CF=0 if cInVal==0 */
```

Comment:
why not use label e.g. `%[a]` instead of number `%2`? using number is error prone.
of course you also need to add label to the input and output section of inline assembly.